ui/tools/problemreporter/problemreporterwidget.cpp

```diff
@@ -0,0 +1,227 @@
+                              textRect.width(),
+                              metrics.height());
+        const QRect descriptionRect = metrics.boundingRect(textRect, Qt::TextWordWrap, description)
+                                             .translated(0, vMargin + titleRect.height());
+
+        const QRect descriptionRect2 = metrics.boundingRect(viewItemOption->rect.adjusted(checkboxRect.width(),0,0,0), Qt::TextWordWrap, description);
+
+        painter->save();
+        painter->drawText(titleRect, Qt::AlignLeft | Qt::AlignVCenter, title);
+        painter->restore();
+        painter->setOpacity(0.5);
+        painter->drawText(descriptionRect, Qt::AlignLeft | Qt::AlignVCenter | Qt::TextWordWrap, description);
+        painter->restore();
+
+
+        style->drawControl(QStyle::CE_CheckBox, &checkboxOption, painter, viewItemOption->widget);
+    }
+
+    QStyleOptionButton copyStyleOptions(QStyleOptionViewItemV4 viewOption, const QModelIndex &index) const
+    {
+        initStyleOption(&viewOption, index);
+        QStyleOptionButton checkboxOption;
+        checkboxOption.direction = viewOption.direction;
+        checkboxOption.fontMetrics = viewOption.fontMetrics;
+        checkboxOption.palette = viewOption.palette;
+        checkboxOption.rect = viewOption.rect;
+        checkboxOption.state = viewOption.state;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+        checkboxOption.styleObject = viewOption.styleObject;
+#endif
+        if (viewOption.checkState == Qt::Checked) {
+            checkboxOption.state |= QStyle::State_On;
+        }
+        return checkboxOption;
+    }
+
+    QSize sizeHint(const QStyleOptionViewItem &option,
+                   const QModelIndex &index) const override
+    {
+        const QStyleOptionViewItemV4 *viewItemOption =
+           qstyleoption_cast<const QStyleOptionViewItemV4 *>(&option); // This is for Qt4 compatibility. Should be a noop in Qt5
+        Q_ASSERT(viewItemOption);
+
+        QStyleOptionButton checkboxOption = copyStyleOptions(*viewItemOption, index);
+
+        const QString title = index.data(Qt::DisplayRole).toString();
+        const QString description = index.data(Qt::ToolTipRole).toString();
+        const QFontMetrics metrics(option.font);
+        const int vMargin = viewItemOption->widget->style()->pixelMetric(QStyle::PM_FocusFrameVMargin);
+        const int hMargin = viewItemOption->widget->style()->pixelMetric(QStyle::PM_FocusFrameHMargin)
+                          + viewItemOption->widget->style()->pixelMetric(QStyle::PM_CheckBoxLabelSpacing);
+        const QRect checkboxRect = viewItemOption->widget->style()->subElementRect(QStyle::SE_CheckBoxIndicator, &checkboxOption, viewItemOption->widget);
+        const QRect titleRect = metrics.boundingRect(title);
+        const int textAvailableWidth = qobject_cast<const QListView*>(viewItemOption->widget)->width() // FIXME this is the wrong size if the view has a horizontal scrollbar...
```

Comment:
unchecked qobject_cast = either should be static_cast, or should have a null check.

Maybe you want to cast to QAbstractScrollArea more generally, and then call viewport()->width()? Not 100% sure.